src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/tasks/payment/PaymentProcessingTask.java

```diff
@@ -0,0 +1,80 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.client.payment.PaymentApiClient;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.payment.Payment;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.payment.Status;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.payment.PaymentService;
+
+import java.util.List;
+
+@Component
+@ConditionalOnProperty(value = "scheduling.task.enabled", matchIfMissing = true)
+public class PaymentProcessingTask {
+
+    private static final Logger log = LoggerFactory.getLogger(PaymentProcessingTask.class);
+    private final PaymentService paymentService;
+    private final PaymentApiClient paymentApiClient;
+    private final int retryCount;
+
+    public PaymentProcessingTask(PaymentService paymentService, PaymentApiClient paymentApiClient,
+                                 @Value("${scheduling.task.post-payments.retry-count}") int retryCount) {
+        this.paymentService = paymentService;
+        this.paymentApiClient = paymentApiClient;
+        this.retryCount = retryCount;
+    }
+
+    @Scheduled(fixedDelayString = "${scheduling.task.post-payments.interval}")
+    public void processPayments() {
+
+        try {
+            List<Payment> payments = paymentService.getPaymentsByStatus(Status.AWAITING.toString());
+
+            if (!payments.isEmpty()) {
+
+                payments.forEach(payment -> {
+
+                    log.info("Posting payment to payment api client for envelope. {}", payment.getEnvelopeId());
+
+                    ResponseEntity<String> responseEntity = postPaymentsToPaymentApi(payment, retryCount);
+
+                    if (responseEntity.getStatusCode().is2xxSuccessful()) {
+
+                        log.info("Posting payment was successful for envelope. {}", payment.getEnvelopeId());
+                        paymentService.updateStatusByEnvelopeId(Status.SUCCESS.toString(), payment.getEnvelopeId());
+                        log.info("Updated payment status to success for envelope. {}", payment.getEnvelopeId());
+                    } else {
+                        log.info("Posting payment was unsuccessful for envelope. {}", payment.getEnvelopeId());
+                        paymentService.updateStatusByEnvelopeId(Status.ERROR.toString(), payment.getEnvelopeId());
+                        log.info("Updated payment status to error for envelope. {}", payment.getEnvelopeId());
+                    }
+                });
+            }
+        } catch (Exception e) {
+            log.info("Error posting payments to payment api client. {}", e.getMessage());
```

Comment:
should be an error
If we have an exception as well, do we want to raise it? Might be a part of the retry logic, so correct me if I am wrong

Reply:
changed

Reply:
added exception in commit [e3b6488](https://github.com/hmcts/bulk-scan-orchestrator/pull/2716/commits/e3b6488bbda7937c27ade02712c42cf5f5dc323d)